quickannotator/api/v1/image/routes.py

```diff
@@ -97,52 +108,57 @@ def post(self, args):
-            file_ext = file_ext[1:]
+            file_ext = str(file_ext[1:]).lower()
+            
             # handle image file
             if file_ext in WSI_extensions:
-                temp_path = fsmanager.nas_write.get_temp_image_path(relative=False)
-                temp_filepath = os.path.join(temp_path,filename)
-
-                # save image to temp folder
-                os.makedirs(temp_path, exist_ok=True)
-                file.save(temp_filepath)
-                
-                # read image info and insert to image table
-                new_image = add_image_by_path(project_id, temp_filepath)
-                # move the actual slides file and update the slide path after create image in DB
-                # image = db_session.query(db_models.Image).filter_by(name=name, path=temp_slide_path).first()
-                image_id = new_image.id
-                slide_folder_path = fsmanager.nas_write.get_project_image_path(project_id, image_id, relative=False)
-                image_full_path = os.path.join(slide_folder_path, filename)
-                # move image file to img_{id} folder
-                os.makedirs(slide_folder_path, exist_ok=True)
-                shutil.move(temp_filepath, image_full_path)
-
-                new_image.path = image_full_path
-                db_session.add(new_image)
-                db_session.commit()
+                image_id = save_image_from_file(project_id, file)
 
+                # get all annotation class name
+                annotation_classes = get_all_annotation_classes()
                 # import annotation if it exist in temp dir
-                for format in AnnotationFileFormats:
-                    temp_image_path = fsmanager.nas_write.get_temp_image_path(relative=False)
-                    annot_filepath = os.path.join(temp_image_path, f'{file_basename}_annotations.{format.value}')
-                    # for geojson
-                    if os.path.exists(annot_filepath):
-                        import_annotations(image_id, annot_filepath)
+                for annot_cls in annotation_classes:
+                    for format in constants.AnnotationFileFormats:
+                        temp_image_path = fsmanager.nas_write.get_temp_image_path(relative=False)
+                        annot_filepath = os.path.join(temp_image_path, f'{file_basename}_{annot_cls.name}_annotations.{format.value}')
+                        # for geojson
+                        if os.path.exists(annot_filepath):
+                            # ?? should always ground true is True ??
+                            import_annotations(image_id, annot_cls.id , True, annot_filepath)
+                
             # handle annotation file
             if file_ext in JSON_extensions:
-                temp_image_path = fsmanager.nas_write.get_temp_image_path(relative=False)
-                annot_filepath = os.path.join(temp_image_path, filename)
+                save_annotation_file_to_temp_dir(file)
```

Comment:
Check if corresponding image has already been added to the database.

If so , call import annotations

Reply:
log if image is found for annotation file.